plone/formwidget/contenttree/source.py

```diff
@@ -154,7 +154,7 @@ def search(self, query, limit=20):
 
         try:
             results = (self.getTermByBrain(brain, real_value=False)
-                       for brain in self.catalog(**catalog_query))
+                       for brain in self.catalog(**catalog_query)[:limit])
```

Comment:
while this works and performance wise only a few would be cutted away (the search may deliver more than limit if exact same items are found - iirc): I don't like the whole code. It returns either a generator or an empty list which is wrong. What about - while at it - writing it like so:

```
    def search(self, query, limit=20):
        catalog_query = self.selectable_filter.criteria.copy()
        catalog_query.update(parse_query(query, self.portal_path))

        if limit and 'sort_limit' not in catalog_query:
            catalog_query['sort_limit'] = limit

        try:            
            for index, brain in enumerate(self.catalog(**catalog_query)):
                if index >= limit:
                    break
                yield self.getTermByBrain(brain, real_value=False)
        except ParseError:
            pass
```